tags/TagMap.md

```diff
@@ -23,23 +23,78 @@ and group stats, annotate traces and logs.
 `TagValue` is a string. It MUST contain only printable ASCII (codes between
 32 and 126)
 
-## TagScope
-
-`TagScope` is used to determine the scope of a `Tag`. The values for the `TagScope` are
-Local or Request. In future, additional values can be added to address specific situations.
-
-The tag creator determines the scope of the tag.
-
-**Local Scope**
-Tag with `Local` scope are used within the process it created. Such tags are not propagated
-across process boundaries. Even if the process is reentrant the tag MUST be excluded from
-propagation when the call leaves the process.
-
-**Request Scope**
-If a tag is created with the `Request` scope then it is propagated across process boundaries subject 
-to outgoing and incoming (on remote side) filter criteria. See `TagPropagationFilter` in 
-[Tag Propagation](#Tag Propagation). Typically 'Request' tags represents a request, processing
- of which may span multiple entities.
+## TagMetadata
+
+`TagMetadata` contains properties associated with a `Tag`. For now only the property `TagTTL`
+is defined. In future, additional properties may be added to address specific situations.
+
+A tag creator determines metadata of a tag it creates.
+
+### TagTTL
+
+`TagTTL` is an integer that represents number of hops a tag can propagate. Anytime a sender serializes a tag,
+sends it over the wire and receiver unserializes the tag then the tag is considered to have travelled one hop. 
+There could be one or more proxy(ies) between sender and receiver. Proxies are treated as transparent
+entities and they do not create additional hops. Every propagation implementation shoud support option 
+`decrementTTL` (default set to true) that allows proxies to set it to false.
+
+**For now, ONLY special values (0 and -1) are supported.**
+
+#### Special Values
+- **NO_PROPAGATION (0)**: Tag with `TagTTL` value of zero is considered to have local scope and
+ is used within the process it created.
+ 
+- **UNLIMITED_PROPAGATION (-1)**: A Tag with `TagTTL` value of -1 can propagate unlimited hops.
+ However, it is still subject to outgoing and incoming (on remote side) filter criteria. 
+ See `TagPropagationFilter` in [Tag Propagation](#Tag Propagation). `TagTTL` value of -1
+ is typical used to represent a request, processing of which may span multiple entities.
+
+#### Example for TagTTL > 0
+On a server side typically there is no information about the caller besides ip/port,
+but in every process there is a notion of "service_name" tag that is added as a "caller" tag before
+serialization when a RPC/HTTP call is made. For the "caller" tag, desirable `TagTTL` value is 1.
+
+Note that TagTTL value of 1 is not supported at this time. The example is listed here simply to
+show a possible use case for TagTTL > 0.
+ 
+### Processing at Receiver and Sender
```

Comment:
Not sure I understand this. Do you suggest that we implement the functionality in the library already? Maybe not a bad idea but we commit on maybe too much for the first phase.

Reply:
I did not mean to implement this right now. However wanted to lay out how TagTTL should be processed when implemented. IMO it is necessary to describe this to make the concept of TagTTL complete. I can make it more clearer that implementation of this processing is not required as there will be nothing on the wire.

Reply:
I take back what I said. Some processing is required on sender and receiver.  For example,
On sender, 

- MUST send the tag without 'TagTTL' if its value is -1. Absence of TagTTL on the wire is treated as having TagTTL of -1.
- MUST not send the tag if the value of `TagTTL` is 0.

On Receiver,
- MUST treat the value of `TagTTL` as -1 if is not present.

The processing has to be part of a tag propagator and not the library. This allows someone not to configure propagator at all if they don't want anything to propagate.